### PR TITLE
arch/arm/src/imxrt: adds support for WDOG1

### PR DESCRIPTION
**Pull request by NicholasChin**
Based on work done by Jake Choy.

## Summary
Adds lower-half driver implementation for Watchdog 1 for IMXRT family.

## Impact
Does not affect existing watchdog code in imxrt_wdog.c

## Testing
Builds clean and checkpatch.sh is also clean

### CODE REVIEW

File: arch/arm/src/imxrt/imxrt_wdog.c [hunk 1]
```diff
@@ -40,28 +40,346 @@
...
+
+#ifdef CONFIG_WATCHDOG_DEVPATH
+#define DEVPATH CONFIG_WATCHDOG_DEVPATH
+#else
+#define DEVPATH "/dev/watchdog0"
+#endif
+
+/* Time out range is from 0 to 128 seconds in 0.5s intervals */
+
+#define WDOG_MIN      (500)   // 0.5 seconds
```
Comment (acassis) [inline arch/arm/src/imxrt/imxrt_wdog.c:73]: Please remove C++ comments

File: arch/arm/src/imxrt/imxrt_wdog.c [hunk 2]
```diff
@@ -40,28 +40,346 @@
...
+#define WDOG_MAX      (128000)   // 128 seconds
+
+#define WDOG_KEEP_ALIVE_KEY1        (0x5555U)
+#define WDOG_KEEP_ALIVE_KEY2        (0xAAAAU)
```
Comment (acassis) [inline arch/arm/src/imxrt/imxrt_wdog.c:77]: Hexa code needs to be in lower case

**Comment by NicholasChin** [comment]
@acassis I've addressed your comments. Thanks!

**Comment by acassis** [comment]
Thank you @NicholasChin

**Pull request merged**